packages/protocol/PaintsCache.ts

```diff
@@ -41,17 +41,18 @@ export async function findFirstMeaningfulPaint() {
       const paint = paints[index];
 
       try {
-        const { data, hash, mimeType } = await screenshotCache.readAsync(
+        const { value } = await StreamingScreenShotCache.readAsync(
           replayClient,
-          paint.point,
-          paint.paintHash
+          paint.time,
+          paint.point
         );
+        if (value && value.hash) {
+          const { width, height } = await getDimensions(value.hash, value.mimeType);
 
-        const { width, height } = await getDimensions(hash, mimeType);
-
-        // Estimate how interesting the image is by the unique pixel density (given an arbitrary image size)
-        if (data.length > (width * height) / 40) {
-          return paint;
+          // Estimate how interesting the image is by the unique pixel density (given an arbitrary image size)
```

Comment:
:question: what does "interesting" mean here? What's special about `40`?

Reply:
You tell me 😄  I didn't write the above code or comment. The code is trying to calculate what % of an image is different pixels, as a proxy for what might be more than just a mostly-blank-screen. That approach– checking the first few frames– doesn't work well though, for apps like Replay, that have a lot of tiny paints as a result of a loading spinner or something. I think we should maybe just ditch this approach, _but_ it seemed out of the scope of this PR so I just left that comment.

Reply:
I'm actually kinda confused - if I pull `main` and look at this file, it looks like you added that code just yesterday in #10273 ?

https://github.com/replayio/devtools/pull/10273/files#diff-dfe818b2f76e449a14505ac73061fa1f303cd25d3a8476466eaf94b5663eb259R52

Reply:
That was a small part of this PR that I landed in advance to make this review smaller. That approach came from here:

https://github.com/replayio/devtools/blob/99212711a2fc7d5b18bd84ae24bfc9a51b52360f/packages/protocol/graphics.ts#L512-L526

It was written three years ago by @jasonLaster (7c98f2a7d27f844e3e9d2f314be5dc2bfc7941f7)